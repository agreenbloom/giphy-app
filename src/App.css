input[type="text"] {
  width: 100%;
  border-radius: 5px;
  height: 35px;
}

button {
  padding: 10px 15px;
  border: solid 1px black;
  background: white;
  margin: 10px 20px 0 0;
  border-radius: 5px;
}

.gallery {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  max-width: 100%;
  margin: 20px;
  align-items: center;
  grid-gap: 10px;
  text-align: center;
}

.gallery img {
  max-width: 100%;
}

.container {
  border: solid 5px black;
  margin: 20px;
  padding: 20px;
  overflow: hidden;
}


@media (max-width: 800px) {
  .gallery{
    grid-template-columns: 1fr 1fr;
  }
}

@media only screen and (max-width: 400px) {
  .gallery {
    grid-template-columns:  1fr;
  }
}
