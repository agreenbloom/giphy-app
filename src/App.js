import React, { Component } from 'react';
import axios from 'axios'

import './App.css';

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      gifs: [],
      store: []
    }
  }


  componentDidMount() {
    axios.get('http://api.giphy.com/v1/gifs/trending?api_key=Mb6qpsLotHxYjpxVHGS9a1WWt0nRcepJ&rating=g')
    .then(resp => console.log(resp.data.data))
  }

  render() {
    return (
      <div className="App">

      </div>
    );
  }
}

export default App;
