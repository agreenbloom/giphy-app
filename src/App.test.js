import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import mockAxios from "axios";

const mockGifs = [
  {
    caption: "happy very funny GIF by Disney Zootopia",
    id:"3NtY188QaxDdC",
    preview: "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
    rating: "g",
    slug: "3NtY188QaxDdC",
    src: "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
    timeTrending: "2017-11-19 23:00:02",
  }, {
    caption:"bird no GIF by Cheezburger",
    id: "g69ZPJfLy7hD2",
    preview: "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
    rating: "g",
    slug: "cheezburger-no-bird-g69ZPJfLy7hD2",
    src: "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
    timeTrending: "2018-08-19 22:45:01",
  }
];

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});


describe('methods', () => {

  describe('handlechange', () => {
    it('should set state of query', () => {
      const wrapper = shallow(<App />);
      const mockValue = 'test';

      wrapper.instance().handleChange(mockValue);
      expect(wrapper.instance().state.query).toBe(mockValue);
    });
  });

  describe('loadMoreGifs', () => {
    it('should offset query by 25 if passed in next', () => {
        const wrapper = shallow(<App />);

        wrapper.instance().loadMoreGifs('next');
        expect(wrapper.instance().state.offset).toBe(25);
    });

    it('should offset query by less than 25 Back', () => {
        const wrapper = shallow(<App />);
        const mockOffset = 75;
        wrapper.setState({
          offset: mockOffset
        })
        wrapper.instance().loadMoreGifs('back');
        expect(wrapper.instance().state.offset).toBe(mockOffset - 25);
    });

    it('should call searchGiphy()', () => {

      const wrapper = shallow(<App />);
      const spy = jest.spyOn(wrapper.instance(), 'searchGiphy');

      wrapper.instance().loadMoreGifs('next');
      expect(wrapper.instance().searchGiphy).toHaveBeenCalled();
    });

  });

  describe('sortByDate', () => {
    it('should sort by trending date', () => {

      const wrapper = shallow(<App />);
      wrapper.setState({
        gifs: mockGifs
      });

      wrapper.instance().sortByDate();
      expect(wrapper.instance().state.gifs[0]).toBe(mockGifs[0])
    });
  });
})
