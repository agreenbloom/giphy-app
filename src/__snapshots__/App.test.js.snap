// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <h1
          className="title"
        >
          Trending Gifs
        </h1>
        <div
          className="inputContainer"
        >
          <Input
            className=""
            disabled={false}
            handleOnEnter={true}
            label="Gifs"
            name="query"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />
          <Button
            disabled={false}
            handleClick={[Function]}
            onClick={null}
          >
            Search
          </Button>
          <Button
            disabled={false}
            handleClick={[Function]}
            onClick={null}
          >
            Sort By Trending Date
          </Button>
        </div>
        <p>
           loading 
        </p>
      </div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Trending Gifs
          </h1>,
          <div
            className="inputContainer"
          >
            <Input
              className=""
              disabled={false}
              handleOnEnter={true}
              label="Gifs"
              name="query"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="text"
              value=""
            />
            <Button
              disabled={false}
              handleClick={[Function]}
              onClick={null}
            >
              Search
            </Button>
            <Button
              disabled={false}
              handleClick={[Function]}
              onClick={null}
            >
              Sort By Trending Date
            </Button>
          </div>,
          <p>
             loading 
          </p>,
          false,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Trending Gifs",
            "className": "title",
          },
          "ref": null,
          "rendered": "Trending Gifs",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                className=""
                disabled={false}
                handleOnEnter={true}
                label="Gifs"
                name="query"
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value=""
              />,
              <Button
                disabled={false}
                handleClick={[Function]}
                onClick={null}
              >
                Search
              </Button>,
              <Button
                disabled={false}
                handleClick={[Function]}
                onClick={null}
              >
                Sort By Trending Date
              </Button>,
            ],
            "className": "inputContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "handleOnEnter": true,
                "label": "Gifs",
                "name": "query",
                "onChange": [Function],
                "onKeyPress": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Search",
                "disabled": false,
                "handleClick": [Function],
                "onClick": null,
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sort By Trending Date",
                "disabled": false,
                "handleClick": [Function],
                "onClick": null,
              },
              "ref": null,
              "rendered": "Sort By Trending Date",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " loading ",
          },
          "ref": null,
          "rendered": " loading ",
          "type": "p",
        },
        false,
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h1
            className="title"
          >
            Trending Gifs
          </h1>
          <div
            className="inputContainer"
          >
            <Input
              className=""
              disabled={false}
              handleOnEnter={true}
              label="Gifs"
              name="query"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="text"
              value=""
            />
            <Button
              disabled={false}
              handleClick={[Function]}
              onClick={null}
            >
              Search
            </Button>
            <Button
              disabled={false}
              handleClick={[Function]}
              onClick={null}
            >
              Sort By Trending Date
            </Button>
          </div>
          <p>
             loading 
          </p>
        </div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
              Trending Gifs
            </h1>,
            <div
              className="inputContainer"
            >
              <Input
                className=""
                disabled={false}
                handleOnEnter={true}
                label="Gifs"
                name="query"
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value=""
              />
              <Button
                disabled={false}
                handleClick={[Function]}
                onClick={null}
              >
                Search
              </Button>
              <Button
                disabled={false}
                handleClick={[Function]}
                onClick={null}
              >
                Sort By Trending Date
              </Button>
            </div>,
            <p>
               loading 
            </p>,
            false,
            false,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Trending Gifs",
              "className": "title",
            },
            "ref": null,
            "rendered": "Trending Gifs",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Input
                  className=""
                  disabled={false}
                  handleOnEnter={true}
                  label="Gifs"
                  name="query"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value=""
                />,
                <Button
                  disabled={false}
                  handleClick={[Function]}
                  onClick={null}
                >
                  Search
                </Button>,
                <Button
                  disabled={false}
                  handleClick={[Function]}
                  onClick={null}
                >
                  Sort By Trending Date
                </Button>,
              ],
              "className": "inputContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "",
                  "disabled": false,
                  "handleOnEnter": true,
                  "label": "Gifs",
                  "name": "query",
                  "onChange": [Function],
                  "onKeyPress": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Search",
                  "disabled": false,
                  "handleClick": [Function],
                  "onClick": null,
                },
                "ref": null,
                "rendered": "Search",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sort By Trending Date",
                  "disabled": false,
                  "handleClick": [Function],
                  "onClick": null,
                },
                "ref": null,
                "rendered": "Sort By Trending Date",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " loading ",
            },
            "ref": null,
            "rendered": " loading ",
            "type": "p",
          },
          false,
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
